### ISSUE TITLE
Safari (Possibly only Mobile) Out-Of-Sync Issues

### ISSUE THREAD

**Issue by jsharkey13** [ucam-cl-dtg/isaac-app#603]
On Safari on iOS (wifi). This issue doesn't affect Chrome on iOS . . .

After logging in, I get a "Session out of Sync" error on every new page reload. But then I remain logged in to the same account when I click continue. I've got no other tabs open, and I haven't attempted to log out or into another account . . .

**Comment by jsharkey13** [comment]
I have finally worked out why this is, and why it now affects Chrome on iOS too!

[Apparently Local Storage can be read but not written](http://stackoverflow.com/questions/14555347/html5-localstorage-error-with-safari-quota-exceeded-err-dom-exception-22-an) in WebKit browsers when Private Browsing is on; there are many reasons why this is [terrible for privacy](https://blog.whatwg.org/this-week-in-html-5-episode-30), but it also makes the user experience on Isaac terrible too.

The checking works by writing the user ID to Local Storage, waiting and then checking the rootScope user ID matches the Local Storage user ID. If they don't match, show an error. *However*, the bad implementation in WebKit means we fail to write the user ID to LocalStorage, then the read clearly returns the wrong thing and we get the pop-up appearing.

To fix this, the [`Persistence.js`](https://github.com/ucam-cl-dtg/isaac-app/blob/master/app/js/app/services/Persistence.js#L23-L57) `save` functions should return `true` if the value was correctly set and `false` if not. Writing to the console doesn't help much on mobile! We should then check that the value has been correctly saved in [`Auth.js`](https://github.com/ucam-cl-dtg/isaac-app/blob/master/app/js/app/services/Auth.js#L133-L151) before we set the interval running! If we failed to write, then print an error to console (in case it's not mobile) and skip checking the user consistency.
